base/ntuple.jl

```diff
@@ -88,3 +88,11 @@ end
         (t..., fill(val, N-M)...)
     end
 end
+
+
+# Specialized extensions for NTuple
+function reverse(t::NTuple{N}) where N
+    ntuple(Val{N}()) do i
```

Comment:
this is better than `Val(N)`?

Reply:
It doesn't matter. Usually `Val(N)` is preferred as it is more concise, but it desugars to effectively the same thing.